ExampleMod/Content/Projectiles/ExampleSpearProjectile.cs

```diff
@@ -0,0 +1,94 @@
+			projectile.friendly = true; // If True, this projectile will hurt enemies.
+		}
+
+		// In here the AI uses this example, to make the code more organized and readable
+		public float movementFactor // Change this value to alter how fast the spear moves
+		{
+			get => projectile.ai[0];
+			set => projectile.ai[0] = value;
+		}
+
+		// It appears that for this AI, only the ai0 field is used!
+		public override void AI() {
+			// Since we access the owner player instance so much, it's useful to create a helper local variable for this
+			// Sadly, Projectile/ModProjectile does not have its own
+			Player projOwner = Main.player[projectile.owner];
+			// Here we set some of the projectile's owner properties, such as held item and itemtime, along with projectile direction and position based on the player
+			Vector2 ownerMountedCenter = projOwner.RotatedRelativePoint(projOwner.MountedCenter, true);
+			projectile.direction = projOwner.direction;
+			projOwner.heldProj = projectile.whoAmI;
+			projOwner.itemTime = projOwner.itemAnimation;
+			projectile.position.X = ownerMountedCenter.X - (float)(projectile.width / 2);
+			projectile.position.Y = ownerMountedCenter.Y - (float)(projectile.height / 2);
+			// As long as the player isn't frozen, the spear can move
+			if (!projOwner.frozen) {
+				if (movementFactor == 0f) // When initially thrown out, the ai0 will be 0f
+				{
+					movementFactor = 3f; // Make sure the spear moves forward when initially thrown out
+					projectile.netUpdate = true; // Make sure to netUpdate this spear
+				}
+				if (projOwner.itemAnimation < projOwner.itemAnimationMax / 3) // Somewhere along the item animation, make sure the spear moves back
+				{
+					movementFactor -= 2.4f;
+				}
+				else // Otherwise, increase the movement factor
+				{
+					movementFactor += 2.1f;
+				}
+			}
+			// Change the spear position based off of the velocity and the movementFactor
+			projectile.position += projectile.velocity * movementFactor;
+			// When we reach the end of the animation, we can kill the spear projectile
+			if (projOwner.itemAnimation == 0) {
+				projectile.Kill();
+			}
+			// Apply proper rotation, with an offset of 135 degrees due to the sprite's rotation, notice the usage of MathHelper, use this class!
+			// MathHelper.ToRadians(xx degrees here)
+			projectile.rotation = projectile.velocity.ToRotation() + MathHelper.ToRadians(135f);
+			// Offset by 90 degrees here
+			if (projectile.spriteDirection == -1) {
+				projectile.rotation -= MathHelper.ToRadians(90f);
+			}
+
+			// These dusts are added later, for the 'ExampleMod' effect
+			if (Main.rand.NextBool(3)) {
+				Dust dust = Dust.NewDustDirect(projectile.position, projectile.height, projectile.width, DustType<Sparkle>(),
+					projectile.velocity.X * .2f, projectile.velocity.Y * .2f, 200, Scale: 1.2f);
+				dust.velocity += projectile.velocity * 0.3f;
+				dust.velocity *= 0.2f;
```

Comment:
maybe you really want to scale down the default -2 to 2 velocity in each direction a projectile gets? in which case you should say that.

Reply:
Im a little confused here by your reply CB, could you elaborate?

Reply:
Just remove these 2 last lines, and adjust velocity through parameters in NewDust. It's confusing shenanigans.

Reply:
`Dust.NewDust` has `SpeedX` and `SpeedY` parameters, which you pass here as `projectile.velocity * .2f`. Inside are the following lines:
```
dust.velocity.X = (float)Main.rand.Next(-20, 21) * 0.1f + SpeedX;
dust.velocity.Y = (float)Main.rand.Next(-20, 21) * 0.1f + SpeedY;
```

As above, adding to the velocity after creating a dust is redundant when you can just pass teh additional velocity to the `SpeedX/Y` parameters. However, all dust come with a random X and Y velocity between -2 and 2 (rand [-20,20] * 0.1). 
If _if_ that's too much random movement/spread and you actually need more stationary dusts, then it makes sense to multiply the dust velocity by a scaling factor after creating it to reduce the random velocity component, but that's not obvious to someone reading this example, so your comment should mention the reasoning.